backend/src/impl/db_utils/system_db_utils.py

```diff
@@ -538,13 +557,11 @@ def db_operations(session: ClientSession) -> bool:
             )
             if not result:
                 abort_with_error_message(400, f"failed to delete system {system_id}")
-            # drop cannot be added to a multi-document transaction, this seems
-            # fine because drop is the last operation. If drop fails, delete
-            # gets rolled back which is our only requirement here.
-            output_collection = DBCollection(
-                db_name=DBUtils.SYSTEM_OUTPUT_DB, collection_name=str(system_id)
-            )
-            DBUtils.drop(output_collection)
+            output_collection = DBUtils.get_system_output_collection(system_id)
+            DBUtils.delete_one_by_id(output_collection, system_id, session=session)
+            for i, analysis_lev in enumerate(sys.system_info.analysis_levels):
+                case_id = f"{system_id}_cases{i}"
+                DBUtils.delete_one_by_id(output_collection, case_id, session=session)
```

Comment:
`case_id` is not the "_id" of these documents. I think this should be something like `delete_many({"data_id": case_id})`